### PR TITLE
Fix python lint errors

### LINKED ISSUE #11542: Python Lint failing on CI, blocking merges
Jobs have begun failing with errors in linting the Python SDK, specifically failing on running `mypy`: https://github.com/pulumi/pulumi/actions/runs/3624968763/jobs/6112540141

Logs:

```
. venv/*/activate && \
	MYPYPATH=./stubs python -m mypy ./lib/pulumi --config-file=mypy.ini
lib/pulumi/runtime/invoke.py:[19](https://github.com/pulumi/pulumi/actions/runs/3624968763/jobs/6112540141#step:16:22)0: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
lib/pulumi/runtime/invoke.py:236: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
lib/pulumi/runtime/invoke.py:251: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
lib/pulumi/runtime/rpc.py:5[21](https://github.com/pulumi/pulumi/actions/runs/3624968763/jobs/6112540141#step:16:24): error: Argument 1 to "FileAsset" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "Union[str, PathLike[Any]]"  [arg-type]
lib/pulumi/runtime/rpc.py:5[23](https://github.com/pulumi/pulumi/actions/runs/3624968763/jobs/6112540141#step:16:26): error: Argument 1 to "StringAsset" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:5[25](https://github.com/pulumi/pulumi/actions/runs/3624968763/jobs/6112540141#step:16:28): error: Argument 1 to "RemoteAsset" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:534: error: Argument 1 to "FileArchive" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:536: error: Argument 1 to "RemoteArchive" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:577: error: Argument 1 to "_parse_urn" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:586: error: Argument 2 to "get_resource_package" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:589: error: Argument 3 to "construct_provider" of "ResourcePackage" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:592: error: Argument 3 to "get_resource_module" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:594: error: Argument 3 to "construct" of "ResourceModule" has incompatible type "Union[Struct, ListValue, str, float, bool, None]"; expected "str"  [arg-type]
lib/pulumi/runtime/rpc.py:601: error: Incompatible return value type (got "Union[Struct, ListValue, str, float, bool, None]", expected "Resource")  [return-value]
```

### PR DESCRIPTION
**Pull request by aq17**
Fixes #11542

### CODE REVIEW

File: sdk/python/lib/pulumi/runtime/rpc.py [hunk 1]
```diff
@@ -571,10 +571,12 @@ def deserialize_properties(
...
 ) -> "Resource":
-    urn = ref_struct["urn"]
-    version = ref_struct["packageVersion"] if "packageVersion" in ref_struct else ""
+    urn = str(ref_struct["urn"])
+    version = (
+        str(ref_struct["packageVersion"]) if "packageVersion" in ref_struct else ""
+    )
 
-    urn_parts = urn_util._parse_urn(urn)
+    urn_parts = urn_util._parse_urn(str(urn))
```
Comment (AaronFriel) [inline sdk/python/lib/pulumi/runtime/rpc.py:579]: This looks redundant with `urn = str(...)`.

```suggestion
    urn_parts = urn_util._parse_urn(urn)
```

File: sdk/python/lib/pulumi/runtime/rpc.py [hunk 2]
```diff
@@ -598,7 +600,8 @@ def deserialize_resource(
         ref_id = ref_struct["id"]
         return deserialize_property(UNKNOWN if ref_id == "" else ref_id, keep_unknowns)
 
-    return urn
+    res = Resource(urn, "urn", False)
```
Comment (justinvp) [inline sdk/python/lib/pulumi/runtime/rpc.py:604]: This is actually changing behavior. We may want to revisit the behavior, but the current intention is to fallback to returning the `urn`. Can we add a lint ignore instead? Or change the type signature of `deserialize_resource` to `Union["Resource", str]` (`Union` will have to be imported from `typing`).

Comment (aq17) [reply sdk/python/lib/pulumi/runtime/rpc.py:604]: I tried simply using `cast(“Resource, urn)` but the type checker still complained. I’ll try the latter first

**Review by justinvp — APPROVED** [review]

**Comment by aq17** [comment]
bors merge p=1

**Pull request merged**